sklearn/metrics/ranking.py

```diff
@@ -100,7 +100,10 @@ def auc(x, y, reorder=False):
                                  "the x array is not increasing: %s" % x)
 
     area = direction * np.trapz(y, x)
-
```

Comment:
The `memmap.sum()` used internally in `trapz` does not convert to float while it does for arrays. Other scorers do not have this issue as highlighted by the test.